packages/expo-firebase-analytics/ios/EXFirebaseAnalytics/EXFirebaseAnalytics.m

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <UMCore/UMUtilities.h>
+#import <EXFirebaseCore/UMFirebaseCoreInterface.h>
+#import <EXFirebaseAnalytics/EXFirebaseAnalytics.h>
+#import <UIKit/UIKit.h>
+#import <Firebase/Firebase.h>
+
+@interface NSObject (Private)
+- (NSString*)_methodDescription;
+@end
+
+@interface EXFirebaseAnalytics ()
+
+@property (nonatomic, weak) UMModuleRegistry *moduleRegistry;
+@property (nonatomic, weak) id<UMFirebaseCoreInterface> firebaseCore;
+
+@end
+
+@implementation EXFirebaseAnalytics
+
+UM_EXPORT_MODULE(ExpoFirebaseAnalytics);
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+- (void)reject:(UMPromiseRejectBlock)reject withException:(NSException *)exception {
+  NSError *error = [NSError errorWithDomain:@"ERR_FIREBASE_ANALYTICS" code:4815162342 userInfo:@{
+        @"message": exception.reason,
+        @"code": exception.name,
+    }];
+  reject(exception.name, exception.reason, error);
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+  _firebaseCore = [moduleRegistry getModuleImplementingProtocol:@protocol(UMFirebaseCoreInterface)];
+}
+
+- (nullable FIRApp *)getAppOrReject:(UMPromiseRejectBlock)reject
+{
+  if (!_firebaseCore) {
+    reject(@"ERR_FIREBASE_ANALYTICS", @"EXFirebaseCore could not be found. Ensure that your app has correctly linked 'expo-firebase-core' and your project has react-native-unimodules installed.", nil);
+    return nil;
+  }
+  FIRApp* defaultApp = [_firebaseCore defaultApp];
```

Comment:
```suggestion
  FIRApp *defaultApp = [_firebaseCore defaultApp];
```

Reply:
Thanks, yep made that mistake everywhere. Fixed it 👍